Issue : Start Player Freezes the app

## Flutter Sound Version : 7.8.4

When play button click flutter app freezes until audio starts playing.
This is happening in only Android app, iOS app doesn't freeze
Can anyone provide the solution?

**Start Player Code**
-----

`Future<void> startPlayer(String url) async {
    try {
      if (url != null) {
        await _ChatState.playerModule.startPlayer(
            fromURI: url,
            codec: Codec.aacMP4,
            whenFinished: () {
              stopPlayer();
            });
      }
    } catch (err) {
      print('error: $err');
    }
  }`

**Flutter Doctor**
------ 

`[✓] Flutter (Channel unknown, 1.22.6, on Mac OS X 10.15.5 19F101 darwin-x64, locale en-GB)
    • Flutter version 1.22.6 at /Volumes/Workspace/Software/flutter
    • Framework revision 9b2d32b605 (8 months ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pavans/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/pavans/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Volumes/Workspace/Software/iOS/xCode/12.1/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    • CocoaPods version 1.10.1

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9306
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)`

Okay, I deleted the clone I had and cloned the entire repository again.

I managed to run both scripts without errors (bin/reldev.sh DEV and bin/flavor.sh LITE)

However, I still can't run the app.

```
FAILURE: Build failed with an exception.

* Where:
Build file 'C:\dev\tau10\tau_sound\android\build.gradle' line: 57

* What went wrong:
A problem occurred evaluating project ':tau_sound'.
> Project with path ':tau_native' could not be found in project ':tau_sound'.
```

--- a/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayer.java
+++ b/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayer.java
@@ -422,7 +422,7 @@ public class FlautoPlayer extends FlautoSession implements MediaPlayer.OnErrorLi
 			{
 
 			}
-			player._play();
+//			player._play();
 			return true;
 	}
 

--- a/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayerMedia.java
+++ b/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayerMedia.java
@@ -37,24 +37,32 @@ class FlautoPlayerMedia extends FlautoPlayerEngineInterface
 	}
 
 	void _startPlayer(String path,  int sampleRate, int numChannels, int blockSize, FlautoPlayer theSession) throws Exception
- 	{
-		this.flautoPlayer = theSession;
- 		mediaPlayer = new MediaPlayer();
+	{
 		if (path == null)
 		{
 			throw new Exception("path is NULL");
 		}
+		this.flautoPlayer = theSession;
+		mediaPlayer = new MediaPlayer();
+
 		mediaPlayer.setDataSource(path);
-		final String pathFile = path;
 		mediaPlayer.setOnPreparedListener(mp -> {flautoPlayer.play(); flautoPlayer.onPrepared();});
 		mediaPlayer.setOnCompletionListener(mp -> flautoPlayer.onCompletion());
-		mediaPlayer.setOnErrorListener(flautoPlayer);
-		mediaPlayer.prepare();
+		mediaPlayer.setOnErrorListener((mp, what, extra) -> {
+			flautoPlayer.onError(mp, what, extra);
+			flautoPlayer.onPrepared();
+			return  false;
+		});
+		mediaPlayer.prepareAsync();
 	}
 
 	void _play()
 	{
-		mediaPlayer.start();
+		mediaPlayer.setOnPreparedListener(mp -> {
+			flautoPlayer.play();
+			flautoPlayer.onPrepared();
+			mp.start();
+		});
 	}
 
 	int feed(byte[] data) throws Exception

--- a/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayerMedia.java
+++ b/android/src/main/java/xyz/canardoux/TauNative/FlautoPlayerMedia.java
@@ -46,7 +46,7 @@ class FlautoPlayerMedia extends FlautoPlayerEngineInterface
 		mediaPlayer = new MediaPlayer();
 
 		mediaPlayer.setDataSource(path);
-		mediaPlayer.setOnPreparedListener(mp -> {flautoPlayer.play(); flautoPlayer.onPrepared();});
+		mediaPlayer.setOnPreparedListener(mp -> {flautoPlayer.play(); _play(); flautoPlayer.onPrepared();});
 		mediaPlayer.setOnCompletionListener(mp -> flautoPlayer.onCompletion());
 		mediaPlayer.setOnErrorListener((mp, what, extra) -> {
 			flautoPlayer.onError(mp, what, extra);
@@ -58,11 +58,7 @@ class FlautoPlayerMedia extends FlautoPlayerEngineInterface
 
 	void _play()
 	{
-		mediaPlayer.setOnPreparedListener(mp -> {
-			flautoPlayer.play();
-			flautoPlayer.onPrepared();
-			mp.start();
-		});
+		mediaPlayer.start();
 	}
 
 	int feed(byte[] data) throws Exception

